pex/docs/server.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2024 Pex project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import errno
+import json
+import logging
+import os
+import re
+import signal
+import subprocess
+import sys
+import time
+from typing import Optional
+
+from pex.common import safe_open
+from pex.typing import TYPE_CHECKING
+from pex.variables import ENV
+from pex.version import __version__
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    import attr  # vendor:skip
+else:
+    from pex.third_party import attr
+
+
+logger = logging.getLogger(__name__)
+
+SERVER_NAME = "Pex v{version} docs HTTP server".format(version=__version__)
+
+_SERVER_DIR = os.path.join(ENV.PEX_ROOT, "docs", "server", __version__)
+
+
+@attr.s(frozen=True)
+class ServerInfo(object):
+    url = attr.ib()  # type: str
+    pid = attr.ib()  # type: int
+
+    def __str__(self):
+        # type: () -> str
+        return "{url} @ {pid}".format(url=self.url, pid=self.pid)
+
+
+@attr.s(frozen=True)
+class Pidfile(object):
```

Comment:
This file was pretty much a pure extraction from `pex.cli.commands.docs` so that `pex --docs` could share the infra.